be/src/vec/exec/format/json/new_json_reader.cpp

```diff
@@ -1007,4 +1015,560 @@ Status NewJsonReader::_read_one_message(std::unique_ptr<uint8_t[]>* file_buf, si
+            // extract root
+            if (_parsed_json_root.size() != 0) {
+                simdjson::ondemand::value val;
+                RETURN_IF_ERROR(JsonFunctions::extract_from_object(cur, _parsed_json_root, &val));
+                // If val is not an object, will throw simdjson::simdjson_error
+                cur = val.get_object();
+            }
+            RETURN_IF_ERROR(_simdjson_write_columns_by_jsonpath(&cur, slot_descs, columns, &valid));
+            // Reached end point
+            if (_array_iter == _array.end()) {
+                // Hint to read next json doc
+                _next_row = _total_rows + 1;
+                break;
+            }
+            ++_array_iter;
+            ++_next_row;
+            if (!valid) {
+                continue; // process next line
+            }
+            *is_empty_row = false;
+            break; // get a valid row, then break
+        } catch (simdjson::simdjson_error& e) {
+            // prevent from endless loop
+            _next_row = _total_rows + 1;
+            fmt::memory_buffer error_msg;
+            fmt::format_to(error_msg, "Parse json data failed. code: {}, error info: {}", e.error(),
+                           e.what());
+            RETURN_IF_ERROR(_state->append_error_msg_to_file(
+                    [&]() -> std::string { return ""; },
+                    [&]() -> std::string { return fmt::to_string(error_msg); }, eof));
+            _counter->num_rows_filtered++;
+            // Before continuing to process other rows, we need to first clean the fail parsed row.
+            for (int i = 0; i < columns.size(); ++i) {
+                if (columns[i]->size() > num_rows) {
+                    columns[i]->pop_back(columns[i]->size() - num_rows);
+                }
+            }
+            if (*_scanner_eof) {
+                // When _scanner_eof is true and valid is false, it means that we have encountered
+                // unqualified data and decided to stop the scan.
+                *is_empty_row = true;
+                return Status::OK();
+            }
+            continue;
+        }
+    } while (_next_row <= _total_rows);
+    return Status::OK();
+}
+
+Status NewJsonReader::_simdjson_handle_nested_complex_json(
+        std::vector<MutableColumnPtr>& columns, const std::vector<SlotDescriptor*>& slot_descs,
+        bool* is_empty_row, bool* eof) {
+    // nested complex json
+    while (true) {
+        size_t num_rows = columns[0]->size();
+        simdjson::ondemand::object cur;
+        try {
+            Status st = _simdjson_parse_json(is_empty_row, eof);
+            if (st.is<DATA_QUALITY_ERROR>()) {
+                continue; // continue to read next
+            }
+            RETURN_IF_ERROR(st);
+            if (*is_empty_row == true) {
+                return Status::OK();
+            }
+            *is_empty_row = false;
+            bool valid = true;
+            cur = _json_value.get_object();
```

Comment:
Is _json_value always an object? If not, is it valid to call get_object()?

Reply:
it will throw expection